components/auth/Join.tsx

```diff
@@ -92,6 +95,21 @@ const Join = () => {
           error={formik.touched.password ? formik.errors.password : undefined}
           onChange={formik.handleChange}
         />
+
+
+        {/* ADDING CHECKBOX HERE */}
+        <InputWithCheckbox
+          type="checkbox"
+          label={t('termsandconditon')}
+          name="agreeToTerms"
+          placeholder={t('termsandcondition')}
+          value={formik.values.agreeToTerms}
+          error={formik.touched.agreeToTerms ? formik.errors.agreeToTerms : undefined}
+          onChange={(e) => {
+            // Use setFieldValue to update formik.values
+            formik.setFieldValue('agreeToTerms', e.target.checked);
+          }}
+        />
```

Comment:
You can use `Checkbox` component directly here, no need to create a new component for it. We'll create a new component for `Checkbox` when we need to add some extra functionality to it.